internal/search/doc/doc.go

```diff
@@ -6,117 +6,31 @@ import (
 	"time"
 
 	"sigs.k8s.io/yaml"
-
-	"google.golang.org/appengine/search"
-)
-
-const (
-	identifierStr   = "identifier"
-	documentStr     = "document"
-	repoURLStr      = "repo_url"
-	filePathStr     = "file_path"
-	creationTimeStr = "creation_time"
 )
 
-// Represents an unbreakable character stream.
-type Atom = search.Atom
-
-// Implements search.FieldLoadSaver in order to index this representation of a kustomization.yaml
-// file.
 type KustomizationDocument struct {
-	identifiers   []Atom
-	FilePath      Atom
-	RepositoryURL Atom
-	DocumentData  string
-	CreationTime  time.Time
+	DocumentData  string    `json:"document,omitempty"`
+	Kinds         []string  `json:"kinds,omitempty"`
+	Identifiers   []string  `json:"identifiers,omitempty"`
```

Comment:
Where and how are `Identifiers` and `Values` used? Do you think changing them to `map[string]string` instead of two string slices work better?

Reply:
This document is directly serialized to elasticsearch which will build indices to support search on each field. It is not meant to meaningfully be used from the server side, where a map would make more sense.

The `Identifiers` support searching for any partial path of an identifier, while the `values` allow searching for a full path + a value.

To my understanding if I represent the values as a structured map, I would lose some of elasticsearch's capabilities since it will not generate the indices and will only be able to do exact text matching for the path, whereas this structure allows it to do fuzzy search, regex, etc. depending on how I formulate the query/allow from the front-end.

So this extra processing will make it easier to find what you're searching for if you have typos, or are missing parts of the path.

I can add comments to better explain that this is what the intended use case of the type is for.

Reply:
Thanks for this explanation. Please add it in the comments.

Reply:
No problem, should have commented it before. Thanks